src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -194,7 +194,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'subnet_name', help='The subnet name.  Creates if creating a new VNet, references if referencing an existing VNet.')
     register_cli_argument(scope, 'admin_password', help='Password for the Virtual Machine if Authentication Type is Password.')
     register_cli_argument(scope, 'ssh_key_value', action=VMSSHFieldAction)
-    register_cli_argument(scope, 'ssh_dest_key_path', completer=FilesCompleter())
+    register_cli_argument(scope, 'ssh_dest_key_path', completer=FilesCompleter(), type=file_type)
```

Comment:
The reason I exposed it is because you might want to store the key in the same location you do on your local machine (because that's the only reason I can think for having the FileCompleter, which was already there). Thus expanding ~ would make sense.